### ISSUE TITLE
Updating or creating a new instance of an extension breaks the instance

### ISSUE THREAD

**Issue by madmacc** [firebase/extensions#681]
### [REQUIRED] Step 2: Describe your configuration

- Extension name: Export Collections to BigQuery firebase/firestore-bigquery-export@0.1.14
- Extension version: 0.1.14

### [REQUIRED] Step 3: Describe the problem

Updating from 0.1.13 to 0.1.14 has an error and breaks the extension.  Now it is not working.
Adding a new extension has the same problem.  

It looks like it is to do with node 8 but as far as I am aware there is no way to upgrade the extensions to node 10.

![firebse-ext-node10-2](https://user-images.githubusercontent.com/43654472/122486945-2d2e1580-d02e-11eb-99a7-381aa4304e52.png)
![firebse-ext-node10](https://user-images.githubusercontent.com/43654472/122486950-2e5f4280-d02e-11eb-82d5-57d1aa1e270c.png)

**Comment by madmacc** [comment]
This is resolved now.
The retry worked for both the second time around.

**Comment by madmacc** [comment]
Actually it is still broken.  I just tried to run it.
![firebse-ext-node10-3](https://user-images.githubusercontent.com/43654472/122493934-a16fb580-d03c-11eb-938e-58e87184071a.png)

**Comment by zdenulo** [comment]
somewhat similar situation. Tried to deploy a new extension Firestore export to BigQuery (version 0.1.14) and have the same issue as initially reported, i.e. Cloud Function is not deployed properly (same cause as reported) thus it's not working

**Comment by rajivsubra1981** [comment]
Yes can confirm that this issue exists. Even if multiple retries allow you to save the extension on the extensions screen, the deployment shows up as failed on the function logs.

**Comment by yeldarby** [comment]
Same here; trying to install the extension on a clean project fails. Full error:

```
ext-firestore-bigquery-export-fsexportbigquery

{
    "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
    "status": {
        "code": 3,
        "message": "
            Build failed: npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
            npm ERR! 
            npm ERR! 
            npm ERR! Missing: @firebaseextensions/firestore-bigquery-change-tracker@^1.1.10
            npm ERR! 

            npm ERR! A complete log of this run can be found in:
            npm ERR! /builder/home/.npm/_logs/2021-06-18T15_46_46_226Z-debug.log; Error ID: beaf8772"
    },
    "authenticationInfo": {
        "principalEmail": "481589474394@cloudservices.gserviceaccount.com"
    },
    "serviceName": "cloudfunctions.googleapis.com",
    "methodName": "google.cloud.functions.v1.CloudFunctionsService.UpdateFunction",
    "resourceName": "projects/{snipped}/locations/us-central1/functions/ext-firestore-bigquery-export-fsexportbigquery"
}
```

**Comment by PesachJTP** [comment]
Hi guys - we're experiencing the same issue. Is there any way I can rollback to version 0.1.13?

**Comment by 747project** [comment]
Is there any way for me to rollback to v0.1.13?

**Comment by ribrdb** [comment]
I was able to reinstall using the firebase cli:
```
firebase ext:install firebase/firestore-bigquery-export@0.1.13
```

**Comment by 747project** [comment]
Thank you @ribrdb

**Comment by madmacc** [comment]
I found another fix if you have multiple instances.

1. Go to cloud functions in GCP: 
https://console.cloud.google.com/functions/list?authuser=1&project=yourproject
2. Click on the extension function
3. Click on edit
4. Click next
5. Change runtime to Node.js 10 from 14
6. Click deploy

![firebse-ext-node10-5](https://user-images.githubusercontent.com/43654472/122624006-747dda00-d0f2-11eb-8af5-0540edf9430e.png)

**Comment by joehan** [comment]
Hey all, thank you for reporting this. I'm currently investigating the issue. For now, either @ribrdb's or @madmacc's solutions should work to get the extension running again ASAP.

**Comment by joehan** [comment]
We've just released 0.1.15 with the node14 upgrade rolled back. If you've been affected by this issue, please update to the latest version and it should be resolved.

We're planning to debug exactly what went wrong with the node14 upgrade, fix it, and rerelease it next week. 

Apologies for the inconvenience everyone! I'm going to close this issue for now & keep an eye on it - if you still have any problems, please let me know!
